ide/project.dependency/src/org/netbeans/modules/project/dependency/ProjectReload.java

```diff
@@ -964,7 +965,7 @@ public static CompletableFuture<ProjectState> withProjectState(Project p, final
         // special case if the state matches & is consistent: if the attempted quality is LESS 
         // than this request's target, the ReloadImplementation might give up 
         if (!doReload && lastKnown.target.isAtLeast(stateRequest.getTargetQuality())) {
-            LOG.log(Level.FINE, "Reload {0}, request: {1}, state {2} - NOOP, finished", new Object[] { p, stateRequest, lastKnown });
+            LOG.log(Level.FINE, "FINISHED: Reload {0}, request: {1}, state {2} - NOOP, finished", new Object[] { p, stateRequest, lastKnown.toString() });
```

Comment:
guard with isLoggable like a few lines below. Since `toString()` of ProjectState is going to do some work now.

Reply:
addressed in 2d925f2